src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -3540,6 +3606,9 @@ void StorageReplicatedMergeTree::dropPartition(const ASTPtr & query, const ASTPt
 
     String partition_id = getPartitionIDFromQuery(partition, query_context);
 
+    if (query_context.getSettingsRef().insert_quorum)
```

Comment:
Why `insert_quorum` setting affects `DROP PARTITION` query? We should check this regardless of the settings.